treeherder/etl/tasks/pulse_tasks.py

```diff
@@ -2,39 +2,48 @@
 This module contains tasks related to pulse job ingestion
 """
 import asyncio
+import logging
 
 import newrelic.agent
 
+from taskcluster.exceptions import TaskclusterRestFailure
 from treeherder.etl.classification_loader import ClassificationLoader
 from treeherder.etl.job_loader import JobLoader
 from treeherder.etl.push_loader import PushLoader
 from treeherder.etl.taskcluster_pulse.handler import handleMessage
 from treeherder.workers.task import retryable_task
 
+logger = logging.getLogger(__name__)
+
 # NOTE: default values for root_url parameters can be removed once all tasks that lack
 # that parameter have been processed
 
-
 @retryable_task(name='store-pulse-tasks', max_retries=10)
 def store_pulse_tasks(
     pulse_job, exchange, routing_key, root_url='https://firefox-ci-tc.services.mozilla.com'
 ):
     """
     Fetches tasks from Taskcluster
     """
+    runs = []
     loop = asyncio.get_event_loop()
     newrelic.agent.add_custom_parameter("exchange", exchange)
     newrelic.agent.add_custom_parameter("routing_key", routing_key)
-    # handleMessage expects messages in this format
-    runs = loop.run_until_complete(
-        handleMessage(
-            {
-                "exchange": exchange,
-                "payload": pulse_job,
-                "root_url": root_url,
-            }
+
+    try:
+        # handleMessage expects messages in this format
+        runs = loop.run_until_complete(
+            handleMessage(
+                {
+                    "exchange": exchange,
+                    "payload": pulse_job,
+                    "root_url": root_url,
+                }
+            )
```

Comment:
if there is an exception, this would most likely result in not ALL messages being parsed, will the remaining unparsed messages be picked up next time this is run (I assume so)?

Previously an exception would terminate the script, now we continue on into the for loop- this is probably good as some subset of messages were handled which would mean there is data in `runs` which need to be processed.

do we need to check for invalid data in `runs`?  I am thinking about the final value which could be corrupt or invalid/unexpected data.

Reply:
Great questions! I'll do some testing locally to check

Reply:
When testing locally, and when I leave it running without hitting the failure so a failing/unknown task gets ignored, I'm not seeing it come back to the failed task.

The `runs` variable is consistently `[]` when it hits a failure here so there's no data that ends up being processed when we fail here.

Based on what you mentioned, and what I've seen locally, the exception seems to be important for coming back to failing tasks? I'm going to see if there's somewhere this failure is bubbling up to that might be failing.